### PR TITLE
add missed await keyword | fix variable reference issue | read version from package.json

### PR DESCRIPTION
**Pull request by ghost**
- add missed await keyword
- fix variable reference issue
- read version from package.json

### CODE REVIEW

**Comment by ghost** [comment]
Regarding #5

File: index.js [hunk 1]
```diff
@@ -461,4 +460,10 @@ async function checkFileExists(extractTo, checkFileName) {
   return files.includes(checkFileName);
 }
 
-getDependenciesAndCheckCompatibility();
+/**
+ * @description
+ * it is possible to utilize top-level `await` since ES13/NodeJs-14.8.0
+ */
+(async function (){
```
Comment (Jordan-Hall) [inline index.js:467]: Let's leave it at top level await

**Comment by Jordan-Hall** [comment]
I think leave as top level other than that changes look right. Nice catch

File: index.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const path = require("path");
 const https = require("https");
 const tar = require('tar');
 const os = require('os');
+const packageJson = require('./package.json');
```
Comment (Jordan-Hall) [inline index.js:8]: Has this been checked locally using npm link to check this works

File: index.js (hunk 1 above)
Comment (ghost) [reply index.js:467]: This make sense as nodejs 14 is being supported since Angular 12

File: index.js (hunk 2 above)
Comment (ghost) [reply index.js:8]: why npm link? I've run `node index.js -v` and it works returning this
obraz

Comment (Jordan-Hall) [reply index.js:8]: Np link crates it as a local package to run rather than locally

**Review by Jordan-Hall — APPROVED** [review]
Looks good if not link works. I assume it will but best to double check

File: index.js (hunk 1 above)
Comment (ghost) [reply index.js:467]: Unfortunately, we cannot leave it at the top-level as this is only available for es modules, not common modules.

Comment (Jordan-Hall) [reply index.js:467]: It's been working fine so should be ok

File: index.js (hunk 2 above)
Comment (ghost) [reply index.js:8]: Yeah, let me check npm link

Comment (ghost) [reply index.js:8]: I've link the repo and run the command in another repo and it looks good

obraz

**Comment by ghost** [comment]
> Looks good if not link works. I assume it will but best to double check

`npm link` works
obraz

**Comment by ghost** [comment]
> I think leave as top level other than that changes look right. Nice catch

It appears to be impossible as top-level await is only available for es modules - I am getting an error when change it to the top-level

**Comment by Jordan-Hall** [comment]
What error because its working fine before this change. Is it because of package.jso. import?

File: index.js (hunk 1 above)
Comment (ghost) [reply index.js:467]: There was no `await` before :)

**Comment by ghost** [comment]
> What error because its working fine before this change. Is it because of package.jso. import?

As I mentioned, there was no `await` before :) And I see `await` as a good pattern in here

**Comment by Jordan-Hall** [comment]
Fair enough. @danielglejzner lools great 1 of 3 issues closed

**Comment by danielglejzner** [comment]
Great stuff ! Thanks @NgDaddy

**Pull request merged**